### PR TITLE
RavenDB-6886 implementation

### PR DESCRIPTION
**Pull request by myarichuk**
* new Raft command IncrementClusterIdentityCommand that will increment identity on Apply() on the leader (quorum)
* generate cluster identity in relevant endoints in BatchHandler and DocumentHandler
* refactor result of Raft command from BlittableJsonReaderObject to object
* if no active transaction in DocumentStorage::ExtractCollectionName() -> throw appropriate exception (prevent NRE)
* make sure that identities are only generated for external requests  …
* because BatchHandler parsed command is a struct, change the way it is updated with identity value
* some relevant unit tests
(RavenDB-6886)

### CODE REVIEW

File: src/Raven.Server/Utils/MiscUtils.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Raven.Server.Utils
+{
+    public static class MiscUtils
+    {
+        /// <summary>
+        /// set longer timespan if debugging, so stuff won't timeout on breakpoints
+        /// </summary>
+        [Conditional("DEBUG")]
+        public static void LongTimespanIfDebugging(ref TimeSpan timespan)
```
Comment (ayende) [inline src/Raven.Server/Utils/MiscUtils.cs:12]: Nice

File: src/Raven.Server/Documents/Handlers/BatchHandler.cs [hunk 2]
```diff
@@ -42,6 +45,18 @@ public class BatchHandler : DatabaseRequestHandler
...
+                for (int i = 0; i < command.ParsedCommands.Count; i++)
+                {
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Type != CommandType.PUT)
+                        continue;
+
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id?.EndsWith("/") != true)
+                        continue;
+
+                    command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id =
+                        await ServerStore.GenerateClusterIdentityAsync(command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id, Database.Name);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/BatchHandler.cs:57]: What do you think about moving identity generation to `BatchRequestParser.ParseAsync` and calling this method once we encounter such a PUT command? We could avoid iterating over the parsed commands here.

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:57]: I'm fine with doing that, but the parser code is tight, and I don't want to introduce async machinery there.

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:57]: Might be better to just have the indexes of all the positions where we need it, and then do it externally.

Comment (arekpalinski) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:57]: That would require the allocation of an additional structure for those indexes/positions to be returned outside `ParseAsync`. Would it be fine?

`ParseAsync` is already contains async machinery due to `await RefillParserBuffer` calls there

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:57]: Hm... okay, so go ahead and do this.

**Pull request merged**